### PR TITLE
Refactor the API

### PR DESCRIPTION
**Pull request by nsarrazin**
# Problem

The API was growing and it was not packaged at all. This PR attempts to provide a (sketch of a) better structure. It also packages the app with poetry properly.

# Solution 
Move folders around, wrap everything in a pyproject.toml, and gets rid of the requirements.txt file.

### CODE REVIEW

File: api/.gitignore [hunk 1]
```diff
@@ -0,0 +1,160 @@
+# Byte-compiled / optimized / DLL files
```
Comment (gaby) [inline api/.gitignore:1]: This file should go one folder up

File: api/src/serge/main.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+from fastapi.staticfiles import StaticFiles
+from starlette.responses import FileResponse
+
+from serge.routers.chat import chat_router
+from serge.routers.model import model_router
+from serge.utils.initiate_database import initiate_database, Settings
+from serge.dependencies import convert_model_files
+
+# Configure logging settings
+logging.basicConfig(
```
Comment (gaby) [inline api/src/serge/main.py:15]: Thoughts on using loguru instead?

https://github.com/Delgan/loguru

File: deploy.sh [hunk 3]
```diff
@@ -4,7 +4,7 @@
 mongod &
```
Comment (gaby) [inline deploy.sh:4]: May be worth moving all these .sh files into a `scripts/` folder. To declutter the root folder

**Review by gaby — CHANGES_REQUESTED** [review]

**Comment by countnoobula** [comment]
With the refactor, is it perhaps valuable to add a set of swagger documentation or openapi JSON to allow us to get easier documentation running?

As user profiles and authentication start to approach, it may be useful for API tokens to tie in with that.

File: api/.gitignore (hunk 1 above)
Comment (nsarrazin) [reply api/.gitignore:1]: I think it makes more sense to keep it there, it's only the python stuff in this gitignore, so I think it belongs in the python server folder.

File: api/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,66 @@
+[tool.poetry]
```
[the remark this answers is not in the document]
Comment (nsarrazin) [reply api/pyproject.toml:1]: Yeah for sure! We do need some tests at some point. Hopefully we can build on top of that to add it.

File: api/src/serge/main.py (hunk 2 above)
Comment (nsarrazin) [reply api/src/serge/main.py:15]: Didn't know about it, it looks really cool. Definitely gonna use it, unless you want to do a PR for it

**Pull request closed without merging**